Auction contracts edits (#172)
tqtezos/minter-sdk: packages/minter-contracts/ligo/src/english_auction/README.md

pauseable_admin_storage %pauseable_admin

```
## Auction State

> - An auction is considered to be `ended` if either the current time is greater than or equal to the end time of the auction or round time has passed after the last bid was placed. 

Reviewer: ```suggestion
- An auction is considered to be `ended` if either the current time is greater than or equal to the end time of the auction or `round_time` has passed after the last bid was placed. 
```

> - An auction is considered to have `started` if the current time is greater than the auctions start time.

Reviewer: ```suggestion
- An auction is considered to have `started` if the current time is greater than or equal to the auction's start time.
```

```

### %bid

> A new bid can be placed if the following confitions are met: 

Reviewer: ```suggestion
A new bid can be placed if the following conditions are met: 
```

## Design Choices/Cautions

1. Only Implicit accounts can place bids.
2. **CAUTION:**   Smart contracts can configure bids. Bidders are encouraged to inspect the code of the selling contract and confirm that the first bid as well as the final reward (upon a call to `Resolve`) will be accepted and not be used to steal gas in the process. See https://www.notion.so/Review-report-for-English-auction-contract-iteration-2-c3610435cc1446d1b6f2b2d60dc86c8e for more details on this.
3. An auction can be configured with an empty asset list theoretically, although this would be a fruitless auction. This was decided to be as such in order to minimuze the cost of configuring an auction. 
4. In the tez auction contracts, "tez guards" are only placed on `cancel`, `configure`, and `resolve` to minimize costs. However, only bid is expected to actually receive tez. Admins should not transfer tez to any other entrypoints to avoid it getting stuck in the contract. 

## Future work

1. Full front running protection.

# English Auction w/ FA2 bids

> In this version of the NFT English Auction contract, bids are made in some FA2 token specified at contract origination as `bid_currency`. Bidders must add the auction contract as an operator for their bid token and then specify the amount they would like to bid in the set FA2 token (`bid_currency`) as an argument to the `bid` entrypoint. The entrypoint will fail if the auction cannot transfer the indicated amount of `bid_currency` to itself from the bidder. In this version of the contract, no tez can be transferred to any entrypoint. An attempt to do so wil fail with `DONT_TRANSFER_TEZ_TO_ANY_ENTRYPOINT`. 

Reviewer: ```suggestion
In this version of the NFT English Auction contract, bids are made in some FA2 token specified at contract origination as `bid_currency`. Bidders must add the auction contract as an operator for their bid token and then specify the amount they would like to bid in the set FA2 token (`bid_currency`) as an argument to the `bid` entrypoint. The entrypoint will fail if the auction cannot transfer the indicated amount of `bid_currency` to itself from the bidder. In this version of the contract, no tez can be transferred to any entrypoint. An attempt to do so will fail with `DONT_TRANSFER_TEZ_TO_ANY_ENTRYPOINT`. 
```
